OpenWhisk actions updated to swift 3

king33 opened the pull request.

tfrank64 commented on BluePic-OpenWhisk/actions/Alchemy.swift:
```diff
@@ -9,34 +9,35 @@ import RestKit
-
-    let alchemyVision = AlchemyVision(apiKey: alchemyKey!)
+    
+    let alchemyKey = args["alchemyKey"] as? String ?? ""
```
So if this ends up being `""` is that ok? Do we just end up with a printed error?

king33 replied:
No, I'll change those assignments to guard statements.

tfrank64 commented on BluePic-OpenWhisk/actions/CloudantRead.swift:
```diff
@@ -1,45 +1,53 @@
     let req = HTTP.request(requestOptions) { response in
         do {
-            str = try response!.readString()!
+            if let response = response {
```
I would just combine these 2 `if let` statements so: `if let response = response, let responseStr = try response.readString() {`

tfrank64 commented on BluePic-OpenWhisk/actions/CloudantWrite.swift:
```diff
@@ -8,50 +8,56 @@ import Foundation
             let req = HTTP.request(requestOptions) { response in
                 do {
-                    str = try response!.readString()!
+                    if let responseUnwrapped = response {
```
Same, combine if let statements

tfrank64 commented on BluePic-OpenWhisk/actions/KituraRequestAuth.swift:
```diff
@@ -8,57 +8,58 @@ import Foundation
+                    
+                    let authData = str.data(using: String.Encoding.utf8, allowLossyConversion: true)
+                    
+                    var authJson: JSON = [:]
```
No need to declare the JSON object here. Just declare 2 lines below

tfrank64 commented on BluePic-OpenWhisk/actions/Orchestrator.swift:
```diff
@@ -8,45 +8,59 @@ import Foundation
+        if let imageURLUnwrapped = imageURL {
+            alchemyInvocation = Whisk.invoke(actionNamed: "/\(targetNamespace)/bluepic/alchemy", withParameters: [
+                "imageURL": "\(imageURLUnwrapped)"
+                ])}
```
Nitpick, bracket in weird place.

tfrank64 commented on BluePic-OpenWhisk/actions/Orchestrator.swift:
```diff
@@ -8,45 +8,59 @@ import Foundation
-    let documentData = documentString.data(using: String.Encoding.utf8, allowLossyConversion: true)!
-    var document = JSON(data: documentData)
-
+    let documentPayload = response["result"] as! [String:Any]
```
I think your best bet for avoiding force unwrapping is to just combine your `if let` statements because they all need to exist anyway for `document`, in this  case, to get data. Something like this:

        var document: JSON = [:]
        if let response = cloudantReadInvocation["response"] as? [String:Any],
        let documentPayload = response["result"] as? [String:Any],
        let documentString:String = documentPayload["document"] as? String,
            let documentData = documentString.data(using: String.Encoding.utf8, allowLossyConversion: true) {
            document = JSON(data: documentData)
        }

king33 replied:
👍

king33 commented:
Thanks @tfrank64 for the helpful comments.

2nd commit addresses your comments and a couple other things:

- I added guard statements at the beginning of the actions so the actions return when required parameters aren't present
- Improved many conditional statements to eliminate some nesting
- Removed all force casting
- fixed a defect in Orchestrator with format of input to weather action

tfrank64 commented:
Changes look great, ready to merge I think.
